src/components/services/UptimeKuma.vue

```diff
@@ -0,0 +1,153 @@
+<template>
+  <Generic :item="item">
+    <template #content>
+      <p class="title is-4">{{ item.name }}</p>
+      <p class="subtitle is-6">
+        <template v-if="item.subtitle">
+          {{ item.subtitle }}
+        </template>
+        <template v-else-if="status">
+          {{ statusMessage }}
+        </template>
+      </p>
+    </template>
+    <template #indicator>
+      <div v-if="status" class="status" :class="status">
+        {{ uptime }}&percnt;
+      </div>
+    </template>
+  </Generic>
+</template>
+
+<script>
+import service from "@/mixins/service.js";
+import Generic from "./Generic.vue";
+
+export default {
+  name: "UptimeKuma",
+  mixins: [service],
+  props: {
+    item: Object,
+  },
+  components: {
+    Generic,
+  },
+  data: () => ({
+    incident: null,
+    heartbeat: null,
+  }),
+  computed: {
+    dashboard: function () {
+      return this.item.slug ? this.item.slug : "default";
+    },
+    status: function () {
+      if (!this.incident) {
+        return "";
+      }
+      return this.incident.incident == null ? this.pageStatus : "bad";
+    },
+    lastHeartBeatList: function () {
+      let result = {};
+
+      for (let id in this.heartbeat.heartbeatList) {
+        let index = this.heartbeat.heartbeatList[id].length - 1;
+        result[id] = this.heartbeat.heartbeatList[id][index];
+      }
+
+      return result;
+    },
+    pageStatus: function () {
+      if (!this.heartbeat) {
+        return "";
+      }
+      if (Object.keys(this.heartbeat.heartbeatList).length === 0) {
+        return "";
+      }
+      let result = "good";
+      let hasUp = false;
+      for (let id in this.lastHeartBeatList) {
+        let beat = this.lastHeartBeatList[id];
+        console.log(beat.status);
+        if (beat.status == 1) {
+          hasUp = true;
+        } else {
+          result = "warn";
+        }
+      }
+      if (!hasUp) {
+        result = "bad";
+      }
+      return result;
+    },
+    statusMessage: function () {
+      if (!this.incident) {
+        return "";
+      }
+      if (this.incident.incident) {
+        return this.incident.incident.title;
+      }
+      return this.pageStatus == "warn"
+        ? "Partially Degraded Service"
+        : "All Systems Operational";
+    },
+    uptime: function () {
+      if (!this.heartbeat) {
+        return 0;
+      }
+      const data = Object.values(this.heartbeat.uptimeList);
+      const percent = data.reduce((a, b) => a + b, 0) / data.length || 0;
+      return (percent * 100).toFixed(1);
+    },
+  },
+  created() {
+    this.item.url = `${this.item.url}/status/${this.dashboard}`;
+    this.fetchStatus();
+  },
+  methods: {
+    fetchStatus: function () {
+      this.fetch(`/api/status-page/${this.dashboard}`)
```

Comment:
Sometimes the Uptime Kuma API will return 304 if nothing has changed. Since we don't store any state or have a local cache, we need to bust the cache with our requests. Easiest way to do that is with a url param. Add here and below

```suggestion
      this.fetch(`/api/status-page/${this.dashboard}?cachebust=${Date.now()}`)
```

Reply:
Does Uptime Kuma do that? It might be a proxy in between since I did not see that happening. Either way, the cache bust is a good idea, thanks!